src/components/chat-button.tsx

```diff
@@ -0,0 +1,107 @@
+'use client'
+
+import React, { useState } from 'react';
+import { MessageSquare, X } from 'lucide-react';
+import { Button } from '@/components/ui/button';
+import { Card, CardContent, CardFooter, CardHeader } from '@/components/ui/card';
+import { Input } from '@/components/ui/input';
+
+const ChatButton = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [messages, setMessages] = useState<{ text: string; isUser: boolean }[]>([]);
+  const [input, setInput] = useState('');
+
+  const toggleChat = () => {
+    setIsOpen(!isOpen);
+    if (!isOpen && messages.length === 0) {
+      setMessages([
+        { text: "Hello! I'm an AI assistant representing Yash Pise. How can I help you today?", isUser: false },
+      ]);
+    }
+  };
+
+  const suggestions = [
+    "Tell me about Yash's experience",
+    "What are Yash's key skills?",
+    "What projects has Yash worked on?",
+  ];
+
+  const handleSend = (message: string = input) => {
+    if (message.trim()) {
+      setMessages([...messages, { text: message, isUser: true }]);
+      // Here you would typically send the message to your AI backend
+      // For now, we'll just simulate a response
+      setTimeout(() => {
+        let response = "I'm sorry, I don't have specific information about that. Is there anything else I can help you with?";
+        if (message.toLowerCase().includes('experience')) {
+          response = "Yash has over 5 years of experience in Data Science and Full Stack Development, working with companies in various industries.";
+        } else if (message.toLowerCase().includes('skills')) {
+          response = "Yash's key skills include Machine Learning, AI, Python, JavaScript, React, and Node.js, among others.";
+        } else if (message.toLowerCase().includes('projects')) {
+          response = "Yash has worked on numerous projects, including AI-powered recommendation systems, data visualization dashboards, and full-stack web applications.";
+        }
+        setMessages(prev => [...prev, { text: response, isUser: false }]);
+      }, 1000);
+      setInput('');
```

Comment:
`suggestion` `non-blocking`:  Maybe this should be a service. It would be good if we implemented dependency injection here to make API consumption easier for next time.